src/Stratis.Bitcoin.Cli/Program.cs

```diff
@@ -35,22 +50,65 @@ public static void Main(string[] args)
+                {
+                    // Process RPC call
+                    try
+                    {
+                        NodeSettings nodeSettings = NodeSettings.FromArguments(args, blockchain, network);
+                        var rpcSettings = new RpcSettings();
+                        rpcSettings.Load(nodeSettings);
+
+                        // Find the binding to 127.0.0.1 or the first available. The logic in RPC settings ensures there will be at least 1.
+                        System.Net.IPEndPoint nodeEndPoint = rpcSettings.Bind.FirstOrDefault(b => b.Address.ToString() == "127.0.0.1") ?? rpcSettings.Bind[0];
+
+                        // Initilize the RPC client with the configured or passed userid, password and endpoint
+                        RPCClient rpc = new RPCClient($"{rpcSettings.RpcUser}:{rpcSettings.RpcPassword}", new Uri($"http://{nodeEndPoint}"));
 
-                // Find the binding to 127.0.0.1 or the first available. The logic in RPC settings ensures there will be at least 1.
-                System.Net.IPEndPoint nodeEndPoint = rpcSettings.Bind.FirstOrDefault(b => b.Address.ToString() == "127.0.0.1") ?? rpcSettings.Bind[0];
+                        // Execute the RPC command
+                        RPCResponse response = rpc.SendCommand(args.ElementAt(1), args.Skip(2).ToArray());
 
-                // Initilize the RPC client with the configured or passed userid, password and endpoint
-                RPCClient rpc = new RPCClient($"{rpcSettings.RpcUser}:{rpcSettings.RpcPassword}", new Uri($"http://{nodeEndPoint}"));
+                        // Return the result as a string to the console
+                        Console.WriteLine(response.ResultString);
+                    }
+                    catch (Exception err)
+                    {
+                        Console.WriteLine(err.Message);
+                    }
+                }
+                else
+                {
+                    // Process API call
 
-                // Execute the RPC command
-                RPCResponse response = rpc.SendCommand(args.ElementAt(1), args.Skip(2).ToArray());
+                    var ApiURI = "http://localhost:37220/";
 
-                // Return the result as a string to the console
-                Console.WriteLine(response.ResultString);
+                    using (var client = new HttpClient())
+                    {
+                        client.DefaultRequestHeaders.Accept.Clear();
+                        client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+                        var url = $"{ApiURI}api/{args.ElementAt(1)}?{string.Join("&", args.Skip(2))}";
+                        try
+                        {
+                            var response = client.GetStringAsync(url).GetAwaiter().GetResult();
+                            // Attempt to format the response
+                            try
```

Comment:
Do we need this `try` and the following `finally`?

Reply:
You are right. All return values are JSON and errors follow a different path. Removed.